docs/configuration.asciidoc

```diff
@@ -0,0 +1,131 @@
+////
+Goal of this doc:
+Provide a complete reference of all available configuration options and where/how they can be set. (Any Elastic-specific configuration options are listed directly. General OpenTelemetry configuration options are linked.)
+
+Assumptions we're comfortable making about the reader:
+* They are familiar with Elastic
+* They are familiar with OpenTelemetry
+////
+
+[[configuration]]
+== Configure
+
+:language: Java
+:language_lc: java
+:distro_name: Elastic Distribution for OpenTelemetry {language}
+
+include::release-status.asciidoc[]
+
+////
+TO DO: How do users set configuration options?
+////
+[discrete]
+[[configure-methods]]
+=== Configuration methods
+
+////
+TO DO:
+Are there multiple ways to set configuration options?
+For example, can they use environment variables OR a config file?
+If there are, what are they and why would you use each method?
+Where/how do you use each method?
+Provide an example to illustrate the syntax.
+////
```

Comment:
> Are there multiple ways to set configuration options?
System properties, Environment variables, or a properties file.

> Where/how do you use each method?
- System properties need to be defined at the JVM start usually on the command-line, and can be seen by anything that can see the executed command-line
  - `java -Dotel.service.name=my-service ...`
- Environment variables need to be defined before the start of the JVM, and can be seen by anything that can see environment variables
  - `export OTEL_SERVICE_NAME=my-service`
  - `java ...`
- properties file needs to be created and populated before the JVM is started
  - `echo otel.service.name=my-service > my.properties`
  - `java -Dotel.javaagent.configuration-file=my.properties ...`